### PR TITLE
Arcbox SQL Automation

### PR DESCRIPTION
**Pull request by DCMattyG**
Added automation to allow onboarding of SQL server for 2 different scenarios:

1) If the Service Principal only has "Contributor" on the target Resource Group, then an icon will be created on the desktop to allow the use to onboard the SQL server post ArcBox deployment. This method will use the login credentials of the actual user, so they themselves will need Microsoft.Authorization/write (Owner/User Access Administrator/Custom Role) on the target Resource Group so that permissions can be modified during the onboarding process.

2) If the Service Principal has Microsoft.Authorization/write on the target Resource Group, the SQL server will automatically be onboarded during the ArcBox deployment, and an assessment will be run.

NOTES: I likely need to include the SQL assessment code for scenario #1, but lets get this tested for now and that can easily be added (and I'll be actively working on this)

### CODE REVIEW

**Comment by dkirby-ms** [comment]
@DCMattyG Can you also please include updates to the READMEs (for full and ITPro) explaining how this works. Right now the READMEs still say that we need UAA on the SPN.

**Comment by dkirby-ms** [comment]
ArcSQL.ps1 should also be renamed to something more descriptive. ArcSQLAutomatedOnboarding.ps1 or something similar.

File: docs/azure_jumpstart_arcbox/flavors/Full/_index.md [hunk 1]
```diff
@@ -322,6 +322,38 @@ ArcBox must be deployed to one of the following regions. Deploying ArcBox outsid
...
+
+- In the event that the Service Principal does *not* have _'Microsoft.Authorization/roleAssignments/write'_ on the target Resource Group, and icon will created on the ArcBox-Client desktop, which will allow you to onboard the Azure Arc-enabled SQL server after the post-deployment automation is complete. To start the onboarding process in this scenario, simply click the _'Onboard SQL Server'_ icon on the desktop. This process should take around 10-15 minutes to complete.
+
+  ![Screenshot showing ArcBox-Client](./sql_manual_onboard_icon.png)
+
+- A pop-up box will walk you through the target SQL server which will be onboarded to Azure Arc,as well as provide details around the flow of the onboarding automation and how to complete the Azure authentication process when prompted.
```
Comment (dkirby-ms) [inline docs/azure_jumpstart_arcbox/flavors/Full/_index.md:333]:
```suggestion
- A pop-up box will walk you through the target SQL server which will be onboarded to Azure Arc, as well as provide details around the flow of the onboarding automation and how to complete the Azure authentication process when prompted.
```

File: docs/azure_jumpstart_arcbox/flavors/Full/_index.md [hunk 2]
```diff
@@ -322,6 +322,38 @@ ArcBox must be deployed to one of the following regions. Deploying ArcBox outsid
...
+
+- You'll then need to provide your Azure credentials to complete the authentication process. The user you login as will need _'Microsoft.Authorization/roleAssignments/write'_ permissions on the ArcBox Resource Group to complete the onboarding process.
+
+  ![Screenshot showing ArcBox-Client](./sql_manual_onboard_login.png)
+
+- The output of each step of the onboarding process will be displayed in the PowerShell script window, so you'll be able to see where the script currently is in the process at all times.
+
+  ![Screenshot showing ArcBox-Client](./sql_manual_onboard_output.png)
+
+- Once complete, you'll receive a pop-up notification informing you that the onboarding process is complete, and to check the Arc blade in the Azure Portal in the next few minutes.
```
Comment (dkirby-ms) [inline docs/azure_jumpstart_arcbox/flavors/Full/_index.md:349]:
```suggestion
- Once complete, you'll receive a pop-up notification informing you that the onboarding process is complete, and to check the Azure Arc blade in the Azure Portal in the next few minutes.
```

**Review by dkirby-ms — CHANGES_REQUESTED** [review]
Some minor suggestions to fix text.

Also, we need to make sure that screenshots for the IT Pro flavor align to the IT Pro flavor. In two of the screenshots for the IT Pro readme updates, I can see that its actually Full flavor being used. In one screen I see Azure Data Studio on the desktop when in IT Pro that should not be there. In another screen, I see the ArcBox RG has 94 resources which tells me its the Full version and not the IT Pro version. This could cause confusion for people that are following along and see a discrepancy from their own RG.

**Comment by DCMattyG** [comment]
@dkirby-ms, I'll work on the suggested changes above including the corrected screenshots today.

**Review by dkirby-ms — APPROVED** [review]

**Pull request merged**